File: src/main/java/jp/developer/bbee/pcassem/domain/migration/MigrationService.java
package jp.developer.bbee.pcassem.domain.migration;

public interface MigrationService {
    /**
     * Firebase IDトークンで認証し、guestIdに紐づくH2データをFirestoreに移行する。
     * 移行済みの場合は何もしない（べき等）。
     */
    void migrate(String idToken, String guestId) throws Exception;
}

File: src/main/java/jp/developer/bbee/pcassem/domain/migration/MigrationService.kt
package jp.developer.bbee.pcassem.domain.migration

interface MigrationService {
    /**
     * Firebase IDトークンで認証し、guestIdに紐づくH2データをFirestoreに移行する。
     * @return true=移行実施, false=移行済みによりスキップ
     */
    @Throws(Exception::class)
    fun migrate(idToken: String, guestId: String): Boolean
}
